File: Molinos.Scato.WebPuerto/src/app/shared/modelos/agente-control-privado.ts

export class AgenteControlPrivado {
      id : number; 
      nombre: string;
      apellido: string;
      name: string; 

      constructor(id, nombre, apellido){
            this.id = id;
            this.nombre = nombre;
            this.apellido = apellido;
            this.name = this.nombre + ' ' + this.apellido;
      }
}

File: Molinos.Scato.Dominio/Entidades/AgenteControlPrivado.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Molinos.Scato.Dominio.Entidades
{
    public class AgenteControlPrivado : IIdentificable
    {
        [Key]
        public virtual int Id { get; set; }
        public virtual string Nombre { get; set; }
        public virtual string Apellido { get; set; }
        public string name { get { return Nombre + ' ' + Apellido; } }
    }
}
